langchain_tools_demo/main.py

```diff
@@ -51,10 +52,11 @@ async def lifespan(app: FastAPI):
 BASE_HISTORY = [{"role": "assistant", "content": "How can I help you?"}]
 
 
-@app.get("/", response_class=HTMLResponse)
+@app.route("/", methods=["GET", "POST"])
 def index(request: Request):
     """Render the default template."""
-    request.session.clear()  # Clear chat history, if needed
+    if request.method == "GET":
+        request.session.clear()  # Do not clear session after login
```

Comment:
We should follow up on this -- it needs to be more intentional to reset history here

Reply:
Probably should add a new endpoint `reset`-- update the session data, removes history, and deletes agent under the hood

Reply:
Addressed in merged PR: https://github.com/GoogleCloudPlatform/genai-databases-retrieval-app/pull/152